runtime/tests/decimal_arith.c

```diff
@@ -178,6 +178,36 @@ void testRem() {
     genAndValidateDecRem("9.00005", "0.0001", "0.00005", 0);
 }
 
+void validateFloatToDec(double val, const char *decStr, PanicCode code) {
+    validate(_bal_decimal_from_float(val), decStr, code);
+}
+
+void testFloatToDec() {
+    validateFloatToDec(0, "0", 0);
+    validateFloatToDec(0.0, "0", 0);
+    validateFloatToDec(1, "1.0", 0);
```

Comment:
It is better to use .0 in all of these so the C type of the literal is double not int.

Reply:
If the result is `0`, we should represent it as `0` according to the subset. All other places are changed to have `.0`.